### ISSUE TITLE
The outputs of `Glob::partitioned` don't interact well with `Glob::walk`.

### ISSUE THREAD

**Issue by olson-sean-k** [olson-sean-k/wax#11]
I need to confirm some of this, but I believe that when `Glob::partitioned` returns an empty `Glob` that `Glob::walk` will never yield any results, even if the prefix is used to derive the root directory and refers to an existing file. For example:

```rust
let path = Path::new(".");
// The glob expression is invariant, so `prefix` is the complete path and `glob` is empty.
let (prefix, glob) = Glob::partitioned("/mnt/media/movie.mp4").unwrap();
for entry in glob.walk(path.join(prefix), usize::MAX) {
    // Should yield `/mnt/media/movie.mp4` if it exists, but does not.
}
```

In the code generated by the [`walk!` macro](https://github.com/olson-sean-k/wax/blob/22c4221bdee7417ed93fa5ebabdda49673460adf/src/lib.rs#L903), the case where there are no remaining path components and also no component regular expressions is not considered. The [`for` loop](https://github.com/olson-sean-k/wax/blob/22c4221bdee7417ed93fa5ebabdda49673460adf/src/lib.rs#L926) is never entered and so no entries are yielded. Again, I haven't had a chance to test this just yet, but I wanted to open an issue since I'm fairly certain this does not work as expected and I don't want to forget. 😅

I believe matching a path using the prefix as seen in [this example](https://github.com/olson-sean-k/wax/blob/22c4221bdee7417ed93fa5ebabdda49673460adf/src/lib.rs#L591) works correctly: stripping the prefix will yield an empty path and an empty glob will only match an empty candidate path.

**Comment by olson-sean-k** [comment]
Confirmed. Invariant glob expressions do not match their corresponding paths. 858d989 has a fix and I'm just waiting for CI to pass before landing it in `master`.

**Comment by olson-sean-k** [comment]
Fixed in 7606c1d. :tada:
